packages/plugin-react/src/index.ts

```diff
@@ -0,0 +1,258 @@
+import { restoreJSX } from './jsx-runtime/restore-jsx'
+
+export interface Options {
+  include?: string | RegExp | Array<string | RegExp>
+  exclude?: string | RegExp | Array<string | RegExp>
+  /**
+   * Enable `react-refresh` integration. Vite disables this in prod env or build mode.
+   * @default true
+   */
+  fastRefresh?: boolean
+  /**
+   * Set this to `"automatic"` to use [vite-react-jsx](https://github.com/alloc/vite-react-jsx).
+   * @default "classic"
+   */
+  jsxRuntime?: 'classic' | 'automatic'
+  /**
+   * Babel configuration applied in both dev and prod.
+   */
+  babel?: TransformOptions
+  /**
+   * @deprecated Use `babel.parserOpts.plugins` instead
+   */
+  parserPlugins?: ParserOptions['plugins']
+}
+
+export default function viteReact(opts: Options = {}): PluginOption[] {
+  let base = '/'
+  let projectRoot = process.cwd()
+  let isProduction = true
+  let skipFastRefresh = opts.fastRefresh === false
+  let filter = createFilter(opts.include, opts.exclude)
+
+  const userPlugins = opts.babel?.plugins || []
+  const userParserPlugins =
+    opts.parserPlugins || opts.babel?.parserOpts?.plugins || []
+
+  const viteBabel: Plugin = {
+    name: 'vite:babel',
+    enforce: 'pre',
+    async transform(code, id, ssr) {
+      if (/\.[tj]sx?$/.test(id)) {
+        const plugins = [...userPlugins]
+
+        const parserPlugins: typeof userParserPlugins = [
+          ...userParserPlugins,
+          'jsx',
+          'importMeta',
+          // This plugin is applied before esbuild transforms the code,
+          // so we need to enable some stage 3 syntax that is supported in
+          // TypeScript and some environments already.
+          'topLevelAwait',
+          'classProperties',
+          'classPrivateProperties',
+          'classPrivateMethods'
+        ]
+
+        const isTypeScript = /\.tsx?$/.test(id)
+        if (isTypeScript) {
+          // TODO: maybe we need to read tsconfig to determine parser plugins to
+          // enable here, but allowing decorators by default since it's very
+          // commonly used with TS.
+          parserPlugins.push('typescript', 'decorators-legacy')
+        }
+
+        const isNodeModules = id.includes('node_modules')
+        if (!isNodeModules && !isProduction) {
+          plugins.push(
+            await interopDefault(
+              import('@babel/plugin-transform-react-jsx-self')
+            ),
+            await interopDefault(
+              import('@babel/plugin-transform-react-jsx-source')
+            )
+          )
+        }
+
+        let useFastRefresh = false
+        if (!skipFastRefresh && !ssr && !isNodeModules) {
+          // Modules with .js or .ts extension must import React.
+          const isReactModule = id.endsWith('x') || code.includes('react')
+          if (isReactModule && filter(id)) {
+            useFastRefresh = true
+            plugins.push([
+              await import('react-refresh/babel'),
```

Comment:
```suggestion
              (await import('react-refresh/babel')).default,
```

Seems that this plugin does require to use the default export or it does not work.